bin/zowe-install-component.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/sh
+DEFAULT_TARGET_DIR=/opt/zowe/extensions
+
+if [[ -z ${ZOWE_ROOT_DIR} ]]
+then
+  export ZOWE_ROOT_DIR=$(cd $(dirname $0)/../;pwd)
+fi
+
+# Try and source the file utils if it exists
+if [[ -f "${ZOWE_ROOT_DIR}/bin/utils/file-utils.sh" ]]
+then
+    . ${ZOWE_ROOT_DIR}/bin/utils/file-utils.sh
+fi
+
+error_handler(){
+    print_error_message "$1"
+    exit 1
+}
+
+extract_to_target_dir(){
+    log_message "Changing directory to $TARGET_DIR"
+
+    cd $TARGET_DIR
+
+    if [ -d "$COMPONENT_FILE" ]; then
+        log_message "Creating symbolic link to the extension's directory."
+        ln -s $COMPONENT_FILE temp-ext-dir
+
+        log_message "Changing directory to $TARGET_DIR/temp-ext-dir"
+        cd temp-ext-dir
+    else
+        # create temporary directory to lay down extension files in
+        log_message "Creating temporary directory to extract extension files into."
+        mkdir temp-ext-dir
+
+        log_message "Changing directory to $TARGET_DIR/temp-ext-dir"
+        cd temp-ext-dir
+
+        log_message "extract file $COMPONENT_FILE"
+
+        if [[ "$COMPONENT_FILE" = *.pax ]]; then
+            pax -ppx -rf "$COMPONENT_FILE"
+        elif [[ "$COMPONENT_FILE" = *.zip ]]; then
+            jar xf "$COMPONENT_FILE"
+        elif [[ "$COMPONENT_FILE" = *.tar ]]; then
+            pax -z tar -xf "$COMPONENT_FILE"
+        fi
+    fi
+
+    # Locate manifest file
+    if [[ -f "manifest.yaml" ]]; then
+        MANIFEST_FILE="manifest.yaml"
+    elif [[ -f "manifest.yml" ]]; then
+        MANIFEST_FILE="manifest.yml"
+    elif [[ -f "manifest.json" ]]; then
+        MANIFEST_FILE="manifest.json"
+    fi
```

Comment:
Seems we need to move some functions into utils scripts. I started working on new utils scripts here https://github.com/zowe/zowe-install-packaging/pull/1822/files#diff-bb1b34e0a3e7bc5990caded4ab189e5aa353bbdf24ae7f16bdd2c9fe848206f2R13. Could be merged soon.